src/NexusMods.App.UI/Controls/GameWidget/GameWidget.axaml.cs

```diff
@@ -10,28 +10,57 @@ namespace NexusMods.App.UI.Controls.GameWidget;
 
 public partial class GameWidget : ReactiveUserControl<IGameWidgetViewModel>
 {
+    private GameWidgetState _previousState;
+
     public GameWidget()
     {
         InitializeComponent();
         this.WhenActivated(d =>
-        {
-            this.OneWayBind(ViewModel, vm => vm.Image, v => v.GameImage.Source)
-                .DisposeWith(d);
-
-            this.WhenAnyValue(view => view.ViewModel!.Image)
-                .WhereNotNull()
-                .OffUi()
-                .Select(BlurAvaloniaImage)
-                .OnUI()
-                .BindToView(this, view => view.BlurryImage.Source)
-                .DisposeWith(d);
-
-            this.OneWayBind(ViewModel, vm => vm.Name, v => v.NameTextBlock.Text)
-                .DisposeWith(d);
-
-            this.BindCommand(ViewModel, vm => vm.PrimaryButton, v => v.PrimaryButton)
-                .DisposeWith(d);
-        });
+            {
+                this.WhenAnyValue(view => view.ViewModel!.State)
+                    .Select(state => state == GameWidgetState.DetectedGame)
+                    .BindToView(this, view => view.DetectedGameStackPanel.IsVisible)
+                    .DisposeWith(d);
+                
+                this.WhenAnyValue(view => view.ViewModel!.State)
+                    .Select(state => state == GameWidgetState.AddingGame)
+                    .BindToView(this, view => view.AddingGameStackPanel.IsVisible)
+                    .DisposeWith(d);
+                
+                this.WhenAnyValue(view => view.ViewModel!.State)
+                    .Select(state => state == GameWidgetState.ManagedGame)
+                    .BindToView(this, view => view.ManagedGameStackPanel.IsVisible)
+                    .DisposeWith(d);
+                
+                this.WhenAnyValue(view => view.ViewModel!.State)
+                    .Select(state => state == GameWidgetState.RemovingGame)
+                    .BindToView(this, view => view.RemovingGameStackPanel.IsVisible)
+                    .DisposeWith(d);
+                
+                this.WhenAnyValue(view => view.ViewModel!.State)
+                    .Select(state => state is GameWidgetState.AddingGame or GameWidgetState.RemovingGame)
+                    .BindToClasses(GameWidgetBorder, "Disabled")
+                    .DisposeWith(d);
```

Comment:
In a perfect world, where subscriptions are cheap, this would be okay. However, subscriptions are sadly more expensive, so this entire block should be one subscription on `State`.